### ISSUE TITLE
paddle.jit.to_static results in wrong gradients

### ISSUE THREAD

**Issue by gdhy9064** [PaddlePaddle/Paddle#43645]
### bug描述 Describe the Bug

使用paddle.jit.to_static会导致反向传播后得到错误的梯度，复现代码如下
```python
import paddle
class MyLayer(paddle.nn.Layer):
    def __init__(self):
        super().__init__()
        self.linear = paddle.nn.Linear(1, 1)

    @paddle.jit.to_static(input_spec=[
        paddle.static.InputSpec(
            shape=[None, None], dtype=paddle.float32
        )
    ])
    def forward(self, x):
        return self.linear(x)

model = MyLayer()
model.clear_gradients()
inp = paddle.ones([1, 1])
out1 = model(inp * 1)
out2 = model(inp * 2)
# loss = out1 * 1 + out2 * 2  # right gradients
loss = out2 * 2 + out1 * 1    # wrong gradients
loss.backward()
print(model.linear.weight.grad)
```
如果使用上面的代码计算得到的仅有的一个参数的梯度为4，如果把`out2 * 2 + out1 * 1`改成`out1 * 1 + out2 * 2`则能得到正确的梯度5

**Comment by Aurelius84** [comment]
感谢您的反馈，我复现下～

**Comment by Aurelius84** [comment]
您好，这个问题已经复现，我们在develop分支已经修复，见：https://github.com/PaddlePaddle/Paddle/pull/43688

**Comment by gdhy9064** [comment]
好的，那我关了这个issue
